From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Johannes Obermayr <johannesobermayr@gmx.de>
Date: Sun, 4 Jun 2023 13:46:13 +0200
Subject: [PATCH] 5418: Add option to force Clang on Linux

Currently Gcc is preferred (not to say forced) when both (Gcc and Clang)
are available on the system. With this patch Clang can be forced via
'gradle -Pclang' switch.

Clang 16.0.4 then failed due to missing '#include <unistd.h>" and an
undefined 'typedef myundef;'.
---
 GPL/nativeBuildProperties.gradle | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/GPL/nativeBuildProperties.gradle b/GPL/nativeBuildProperties.gradle
index 23eb20a824..ebdc4cd5c4 100644
--- a/GPL/nativeBuildProperties.gradle
+++ b/GPL/nativeBuildProperties.gradle
@@ -41,8 +41,12 @@ model {
 		def current = getCurrentPlatformName()
 		if (isLinux(current)) {
 			// https://github.com/gradle/gradle/issues/17660 (fixed in Gradle 8.11)
-			gcc(Gcc).target(current)
-			clang(Clang).target(current)
+			if (project.hasProperty('clang')) {
+				clang(Clang).target(current)
+			}
+			gcc(Gcc) {
+				gcc(Gcc).target(current)
+			}
 		}
 		if (isFreeBSD(current)) {
 			// https://github.com/gradle/gradle/issues/32895
-- 
2.45.1

